docs/android/tips/dependencies.md

```diff
@@ -0,0 +1,57 @@
+---
+title: "Android Dependencies"
+sidebar_label: "Dependencies"
+---
+
+## Version Catalog
+
+We use the [version catalog](https://docs.gradle.org/current/userguide/version_catalogs.html) to manage all libraries directly used in the project. This is the **only allowed method** for adding dependencies. Adding a dependency outside of the catalog is strictly forbidden to maintain consistency and traceability.
+
+### Benefits of Using a Version Catalog
+
+- **Centralized Management**: All dependencies are defined in one place (`gradle/libs.versions.toml`), making it easier to track and update them.
+- **Consistency**: Ensures that all modules use the same versions of shared dependencies.
+- **Simplified Updates**: Makes it easier to update dependencies across the entire project.
+
+## Managing Dependencies and Lockfiles
+
+This project utilizes Gradle's [dependency locking](https://docs.gradle.org/current/userguide/dependency_locking.html) feature to ensure consistent and reproducible builds by tracking the precise versions of all libraries used.
+
+### Why Use Dependency Locking
+
+- **Reproducible Builds**: Ensures that builds are consistent across different environments by locking the exact versions of all dependencies.
+- **Avoids Surprises**: Prevents unexpected updates to transitive dependencies that could break the build.
+
+### Updating Dependencies and Lockfiles
+
+When adding or updating a dependency in `gradle/libs.versions.toml`, it's crucial to also update the corresponding lockfiles. The lockfiles capture the exact versions of all direct and transitive dependencies.
```

Comment:
Should we also mention the CI will fail if the lock files are not updated when the PR is submitted here?